rfcs/support_virtual_test_suites.md

```diff
@@ -0,0 +1,47 @@
+# RFC 109: support virtual test suites in wptrunner
+
+## Summary
+
+Support virtual test suites in wptrunner, so that we can run tests with
+additional browser flags to exercise different code paths.
+
+## Details
+
+Add an *optional* command line argument to pass in a json configuration file. The
+content of the configuration file is defined by each vendor, and defines a list of
+virtual test suites. Each virtual test suite has a prefix, and a list of bases
+and args. For example, you could define a (hypothetical) virtual test suite
+using the following json configuration:
+
+```json
+{
+  "prefix": "css-highlight",
+  "bases": ["css/css-pseudo", "css/css-highlight-api"],
+  "args": ["--enable-features=css-highlight"]
```

Comment:
I think the virtual tests are fundamentally vendor specific. That's why I am saying we need a command line argument to pass in the config. Each vendor should have their own config file staying outside of wpt.

Reply:
What would be a better way here? I am not sure how do we enable/disable features in other browsers.

Reply:
Well for gecko we'd set specific prefs. I think you should ensure the mechanism is general enough that it can adapt to different possible ways of actually specifying the configuration. For example it could be specified in terms of equivalent additional args to wptrunner itself; that allows passing in args to the binary, prefs, args to any webdriver wrapper, etc.

Reply:
Thanks. I think I will be flexible here. As long as it is doable I will be fine.

James, can you propose a format here? Maybe we can make it a key:value list? e.g. ["arg": "--enable-features=feature-a", "pref": "some-pref"]?

How do we pass prefs to the browser? Through environments?